website/content/en/docs/best-practices/multi-tenancy.md

```diff
@@ -0,0 +1,162 @@
+in production. In such cases your operator should minimally **have the option to prevent [NetworkPolicy][k8s-network-policy] objects from being
+created** and leave these to the user. In more advanced cases, your operator should create [NetworkPolicy][k8s-network-policy] configurations that
+follow the least-privilege principle as well implement deny-all-allow-some rules.
+
+### Managing Resources
+
+#### Resource Requests
+
+Create resource requests for CPU and memory:
+
+* for your Operator's manager `Deployment` (See `config/manager/manager.yaml`)
+* for any `Pods` and `Deployments` which are created by your Operator
+
+```yml
+  ...
+  # TODO(user): Configure the resources accordingly based on the project requirements.
+  # More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+  resources:
+    requests:
+      cpu: 10m
+      memory: 64Mi
+    ... 
+```
+
+In order to prevent tenants from consuming all the resources on a cluster or blocking other workloads 
+from being scheduled, many production clusters will define [ResourceQuota][k8s-resources-quotas] configuration.
+[ResourceQuota][k8s-resources-quotas] configuration in a namespace will cause all pods without 
+[Requests and limits][k8s-requests-limits] for `CPU` and `Memory` to be rejected.
+
+Because OLM-managed Operators do not run on Cluster infra nodes, they will consume paid CPU cores. 
+Users will naturally want to limit the maximum amount of resources these non-workload components can take up. 
+For this reason, if you are integrating with [OLM][olm], you should, **always ship your operator `Deployment` with resource requests amounts for CPU and memory**.
+
+The [ResourceQuota][k8s-resources-quotas] configuration also is applied to tenant workloads that are managed by your operator. Cluster administrators will typically set
+a [ResourceQuota][k8s-resources-quotas] for each tenant's namespace/project as part of the onBoarding. If your operator creates `Pods/Deployments` inside
+the tenant namespace which do not specify at least resource requests for `CPU` and `Memory`, those `Pods/Deployments` will fail to schedule and tenants
+will not be able to use your operator. Therefore, your operator should **always apply at least resource requests for CPU and
+memory** for `Pods/Deployments` that it creates as part of reconciliation. Ideally your operator also applies resource 
+limits to those `Pods/Deployments`. 
+
+#### General guidelines
+
+**Note that pretty much all controllers consume:**
+- `CPU`: largely based on the number of reconciliations they perform, which are generally related to event activity for resources they're watching.
+- `Memory`: largely based on the number of primary resources that exist (multiplied by some factor based on the number of operand resources they need to watch as a result) via informer caches.
```

Comment:
I would rephrase it. I think that one of the pitfalls that many operators are failing into is that they watch resources with high cardinality like secrets possibly in all namespaces. Besides the security aspect, which may be worth mentioning somewhere this has a massive impact on the memory used by the controller on big clusters. Such resources should either be filtered by label or possibly avoided by for instance using a custom resource instead.

Reply:
Also it may be worth mentioning ResyncPeriod. Setting it to low would also cause the unnecessary consumption of CPU cycles

Reply:
The points are valid, however, they do not fit in the scope context of this doc. 
IHMO would be nice we have a doc about good practices that describe them and how to avoid a huge unnecessarily consumption of the resources. Please, feel to work on that., I think would be a great follow up.

Reply:
Ack. I created: https://github.com/operator-framework/operator-sdk/issues/5382

Reply:
Cool closing that as sorted out